ingenialink/configuration_file.py

```diff
@@ -0,0 +1,458 @@
+        }
+        self.dtype_value_to_str = {
+            value: key for key, value in Dictionary.dtype_xdf_options.items()
+        }
+
+    @classmethod
+    def from_xcf(cls, element: ElementTree.Element) -> "ConfigRegister":
+        """Creates a register from register XML element.
+
+        Returns:
+            ConfigRegister filled with XML element data
+
+        Raises:
+            ValueError: wrong fields type
+            KeyError: an attribute is missing
+        """
+        uid = element.attrib[cls.ID_ATTR]
+        subnode = int(element.attrib[cls.SUBNODE_ATTR])
+        dtype = Dictionary.dtype_xdf_options[element.attrib[cls.DTYPE_ATTR]]
+        access = Dictionary.access_xdf_options[element.attrib[cls.ACCESS_ATTR]]
+        storage: Union[float, int, str, bool]
+        if dtype == RegDtype.FLOAT:
+            storage = float(element.attrib[cls.STORAGE_ATTR])
+        elif dtype in [
+            RegDtype.S8,
+            RegDtype.U8,
+            RegDtype.S16,
+            RegDtype.U16,
+            RegDtype.S32,
+            RegDtype.U32,
+            RegDtype.S64,
+            RegDtype.U64,
+        ]:
+            storage = int(element.attrib[cls.STORAGE_ATTR])
+        elif dtype == RegDtype.STR:
+            storage = element.attrib[cls.STORAGE_ATTR]
+        elif dtype == RegDtype.BOOL:
+            storage = bool(element.attrib[cls.STORAGE_ATTR])
+        else:
+            raise NotImplementedError
+        return cls(uid, subnode, dtype, access, storage)
+
+    @classmethod
+    def from_register(
+        cls, register: Register, value: Union[float, int, str, bool]
+    ) -> "ConfigRegister":
+        """Creates a ConfigRegister from a register, and it's value.
+
+        Returns:
+            ConfigRegister instance filled with register data
+
+        Raises:
+            ValueError: Register has no an identifier
+        """
+        if register.identifier is None:
+            raise ValueError("register has not an identifier")
+        return cls(register.identifier, register.subnode, register.dtype, register.access, value)
+
+    def to_xcf(self) -> ElementTree.Element:
+        """Creates a XMl element from class data.
+
+        Returns:
+            XML register element filled with class data
+        """
+        register_xml = ElementTree.Element(self.ELEMENT_NAME)
+        register_xml.set(self.ACCESS_ATTR, self.access_value_to_str[self.access])
+        register_xml.set(self.DTYPE_ATTR, self.dtype_value_to_str[self.dtype])
+        register_xml.set(self.ID_ATTR, self.uid)
+        register_xml.set(self.SUBNODE_ATTR, str(self.subnode))
+        if isinstance(self.storage, float):
+            register_xml.set(self.STORAGE_ATTR, str(np.float32(self.storage)))
+        else:
+            register_xml.set(self.STORAGE_ATTR, str(self.storage))
+        return register_xml
+
+
+class XMLBase(ABC):
+    """Base class to manipulate XML files."""
```

Comment:
This is only used here once, is the intention to use it as a base for dictionary parsers?

Reply:
Nice! No rush for this, create issue so we don't forget